### PR TITLE
added support for different deviceTypes

### PR DESCRIPTION
**Pull request by Zefau**

### CODE REVIEW

File: lib/bridge.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ Bridge.prototype.list = function list () {
       return nukis;
     })
     .map(function processEachNuki (nuki) {
-      nuki.nuki = new Nuki(self, nuki.nukiId, {delayBetweenRequests: self.delayBetweenRequests});
+      nuki.nuki = new Nuki(self, nuki.nukiId, nuki.deviceType || 0);
```
Comment (Mik13) [inline lib/bridge.js]: So I guess `deviceType` is coming from the list, right?

File: lib/bridge.js [hunk 2]
```diff
@@ -101,7 +101,7 @@ Bridge.prototype.get = function get (nukiId, doNotValidate) {
   var self = this;
 
   if (doNotValidate) {
-    return Promise.resolve(new Nuki(self, nukiId, {delayBetweenRequests: self.delayBetweenRequests}));
+    return Promise.resolve(new Nuki(self, nukiId, nuki.deviceType || 0));
```
Comment (Mik13) [inline lib/bridge.js]: `nuki` is not defined here

File: lib/bridge.js [hunk 3]
```diff
@@ -119,7 +119,7 @@ Bridge.prototype.get = function get (nukiId, doNotValidate) {
         throw new Error('did not find nuki!');
       }
 
-      return new Nuki(self, nukiId, {delayBetweenRequests: self.delayBetweenRequests});
+      return new Nuki(self, nukiId, nuki.deviceType || 0);
```
Comment (Mik13) [inline lib/bridge.js]: Do I overlook something, seems like it isn't here too

File: lib/bridge.js (hunk 1 above)
Comment (Zefau) [reply lib/bridge.js]: Yes, `deviceType` is retrieved from `/list` as documented in the updated Nuki Bridge API docs, see https://developer.nuki.io/page/nuki-bridge-http-api-190/4#heading--list.
- 0 => smartlock (Nuki Smart Lock)
- 2 => opener (Nuki Opener)

File: lib/bridge.js (hunk 2 above)
Comment (Zefau) [reply lib/bridge.js]: True. Means we could either
1. remove `doNotValidate` and retrieve the `deviceType` in the course of the validation
OR
2. we add an additional parameter `deviceType`.

I would prefer the first option. What do you think?

**Pull request closed without merging**